### PR TITLE
made cirrus build scripts consistent

### PR DESCRIPTION
**Pull request by Sertonix**
This makes the cirrus build scripts more consistent. It changes the names of the scripts to be consistent and it puts the scripts into a consistend order (`prepare` `install` `build` `build_binary` `test`).

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ linux_task:
   container:
     image: node:16-slim
     memory: 8G
-  test_script:
+  prepair_script:
```
Comment (Spiker985) [inline .cirrus.yml:6]:
```suggestion
  prepare_script:
```

File: .cirrus.yml [hunk 2]
```diff
@@ -46,7 +49,7 @@ arm_linux_task:
     memory: 8G
   env:
     USE_SYSTEM_FPM: 'true'
-  prepare_script:
+  prepair_script:
```
Comment (Spiker985) [inline .cirrus.yml:52]:
```suggestion
  prepare_script:
```

File: .cirrus.yml [hunk 3]
```diff
@@ -98,16 +103,26 @@ silicon_mac_task:
   macos_instance:
     image: ghcr.io/cirruslabs/macos-monterey-base:latest
     memory: 8G
-  test_script:
+  prepair_script:
```
Comment (Spiker985) [inline .cirrus.yml:106]:
```suggestion
  prepare_script:
```

File: .cirrus.yml [hunk 4]
```diff
@@ -116,18 +131,13 @@ silicon_mac_task:
...
-  binary_artifacts:
-    path: ./binaries/*
 
 intel_mac_task:
   alias: mac
   macos_instance:
     image: ghcr.io/cirruslabs/macos-monterey-base:latest
     memory: 8G
-  dist_script:
+  prepair_script:
```
Comment (Spiker985) [inline .cirrus.yml:140]:
```suggestion
  prepare_script:
```

File: .cirrus.yml [hunk 5]
```diff
@@ -163,27 +178,18 @@ windows_task:
   env:
     CIRRUS_SHELL: bash
     PATH: C:\Python310\Scripts\;C:\Python310\;%PATH%;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\Users\User\AppData\Local\Microsoft\WindowsApps;C:\Users\User\AppData\Roaming\npm;C:\Program Files\Microsoft Visual Studio\2022\Community\Msbuild\Current\Bin\
-  install_deps_script:
+  prepair_script:
```
Comment (Spiker985) [inline .cirrus.yml:181]:
```suggestion
  prepare_script:
```

**Review by Spiker985 — CHANGES_REQUESTED** [review]
The addition of more labels doesn't impact the availability of resources/env vars/etc between instructions, right? It doesn't act like adding an additional job within a GH action?
@mauricioszabo

**Comment by mauricioszabo** [comment]
It can impact - it seems that on Silicon Mac it is impacting, so we may need to re-export PATH on the new label

**Comment by Spiker985** [comment]
> It can impact - it seems that on Silicon Mac it is impacting, so we may need to re-export PATH on the new label

It looks like the PATH _is_ being re-exported per label for Macs. [L114](https://github.com/pulsar-edit/pulsar/pull/239/files?diff=unified&w=0#diff-62587956f943bb2503db7bc6dd27d0d888074a1c0ecaab3f570ad611aff0f7bbR114), [L117](https://github.com/pulsar-edit/pulsar/pull/239/files?diff=unified&w=0#diff-62587956f943bb2503db7bc6dd27d0d888074a1c0ecaab3f570ad611aff0f7bbR117), [L121](https://github.com/pulsar-edit/pulsar/pull/239/files?diff=unified&w=0#diff-62587956f943bb2503db7bc6dd27d0d888074a1c0ecaab3f570ad611aff0f7bbR121) and [L152](https://github.com/pulsar-edit/pulsar/pull/239/files?diff=unified&w=0#diff-62587956f943bb2503db7bc6dd27d0d888074a1c0ecaab3f570ad611aff0f7bbR152), [L155](https://github.com/pulsar-edit/pulsar/pull/239/files?diff=unified&w=0#diff-62587956f943bb2503db7bc6dd27d0d888074a1c0ecaab3f570ad611aff0f7bbR155), [L159](https://github.com/pulsar-edit/pulsar/pull/239/files?diff=unified&w=0#diff-62587956f943bb2503db7bc6dd27d0d888074a1c0ecaab3f570ad611aff0f7bbR159)

Unless you're referring to something else?

**Review by Spiker985 — APPROVED** [review]

File: .cirrus.yml [hunk 6]
```diff
@@ -98,16 +103,26 @@ silicon_mac_task:
...
+    - export PATH="/opt/homebrew/bin:/opt/homebrew/opt/node@16/bin:$PATH"
     - yarn build
     - yarn run build:apm
+  build_binary_script:
+    - export PATH="/opt/homebrew/bin:/opt/homebrew/opt/node@16/bin:$PATH"
+    - yarn dist || yarn dist
+  binary_artifacts:
+    path: ./binaries/*
+  test_script:
     - PLAYWRIGHT_JUNIT_OUTPUT_NAME=report.xml npx playwright test --reporter=junit,list
```
Comment (DeeDeeG) [inline .cirrus.yml:126]: This is per @mauricioszabo's suggestion in a comment above...

```suggestion
  test_script:
    - export PATH="/opt/homebrew/bin:/opt/homebrew/opt/node@16/bin:$PATH"
    - PLAYWRIGHT_JUNIT_OUTPUT_NAME=report.xml npx playwright test --reporter=junit,list
```

Hopefully this fixes the test failure? ("`npx: command not found`")

<details><summary>Full error message:  (click to expand):</summary>

```console
PLAYWRIGHT_JUNIT_OUTPUT_NAME=report.xml npx playwright test --reporter=junit,list
/var/folders/76/zy5ktkns50v6gt5g8r0sf6sc0000gn/T/scriptseb9d18a44784045d87f3c67cf22746e9.sh: line 5: npx: command not found
Exit status: 127
```

</details>

**Comment by DeeDeeG** [comment]
@Sertonix do you mind if I fix the merge conflicts here (and push the merge fix to this branch) and then merge this PR? Or if you want to fix the merge conflicts yourself, I can let you do it.

I would be a second approve for this, since it makes it easier to debug a CI run for any OS even if you've only looked at runs for a different OS before. Also, makes it much easier to keep the different OSes in sync with one-another.

We just have to make it synced with recent .cirrus.yml changes form other PRs.

Thanks for doing this PR in the first place, @Sertonix!

**Comment by Sertonix** [comment]
@DeeDeeG you can fix the merge conflicts.

**Comment by DeeDeeG** [comment]
I would want to merge this just after the Regular 1.103 release anticipated later today. So we can shake things up _after_ a Regular release, not just before/in the middle of it. But yeah, I'd like to merge this then.

**Review by DeeDeeG — APPROVED** [review]
Looks good to me. Makes the names and order of the steps consistent for each OS/Arch.

Deletes some commented out steps for Windows and the "Upload Videos Artifacts" step on Windows, which there are currently no videos to upload, since the tests aren't run on Windows...

Approving, will merge soon. Thanks for this! Sorry it took a while to review.

**Pull request merged**